mapie/tests/test_regression.py

```diff
@@ -502,30 +503,38 @@ def test_aggregate_with_mask_with_prefit() -> None:
     """
     Test ``_aggregate_with_mask`` in case ``cv`` is ``"prefit"``.
     """
-    mapie_reg = MapieRegressor(cv="prefit")
+    mapie_reg = MapieRegressor(LinearRegression().fit(X, y), cv="prefit")
+    mapie_reg = mapie_reg.fit(X, y)
     with pytest.raises(
         ValueError,
         match=r".*There should not be aggregation of predictions if cv is*",
     ):
-        mapie_reg._aggregate_with_mask(k, k)
+        mapie_reg.estimator_._aggregate_with_mask(k, k)
 
-    mapie_reg = MapieRegressor(agg_function="nonsense")
+
+def test_aggregate_with_mask_with_invalid_agg_function() -> None:
+    """Test ``_aggregate_with_mask`` in case ``agg_function`` is invalid."""
+    ens_reg = EnsembleRegressor(LinearRegression(), "plus",
+                                KFold(
+                                    n_splits=5, random_state=None, shuffle=True
+                                ),
+                                "nonsense", None, random_state, 0.20, False
+                                )
     with pytest.raises(
         ValueError,
         match=r".*The value of self.agg_function is not correct*",
     ):
-        mapie_reg._aggregate_with_mask(k, k)
+        ens_reg._aggregate_with_mask(k, k)
 
 
 def test_pred_loof_isnan() -> None:
     """Test that if validation set is empty then prediction is empty."""
     mapie_reg = MapieRegressor()
-    y_pred: ArrayLike
-    _, y_pred, _ = mapie_reg._fit_and_predict_oof_model(
+    mapie_reg = mapie_reg.fit(X, y)
+    y_pred: NDArray
+    y_pred, _ = mapie_reg.estimator_._predict_oof_estimator(
```

Comment:
UPD: with my previous consideration, prefer to use `fit_and_predict` method to do that.